include/openthread/openthread_enable_defines.h

```diff
@@ -0,0 +1,48 @@
+/*
+ *  Copyright (c) 2017, The OpenThread Authors.
```

Comment:
Maybe we should rename this file, I think this file should be named: openthread_config.h

My objection here is the "enables define" name just seems weird/out of place.

the *GENERATED* file should be come:  openthread-config-generated.h

Then platform that need to use a specific configuration can provide their own

Something like this:
```
#if  !defined(OPENTHREAD_CONFIG_FILE)
#define OPENTHREAD_CONFIG_FILE  "openthread-config-generated.h"
#endif

#include OPENTHREAD_CONFIG_FILE
```

Then, as platforms